src/tests/ftest/launch.py

```diff
@@ -346,7 +347,10 @@ def set_provider_environment(interface, args):
         interface (str): the current interface being used.
     """
     # Use the detected provider if one is not set
-    provider = os.environ.get("CRT_PHY_ADDR_STR")
+    if args.provider:
+        provider = args.provider
+    else:
+        provider = os.environ.get("CRT_PHY_ADDR_STR")
```

Comment:
Just FYI, I've been enjoying this syntax lately:
```suggestion
    provider = args.provider or os.environ.get("CRT_PHY_ADDR_STR")
```

Reply:
I like that.  I have a few other PRs with launch.py changes.  I'll make the update in one of those.